system/jlib/jutil.cpp

```diff
@@ -1692,6 +1692,20 @@ void doStackProbe()
 #pragma GCC diagnostic pop
 #endif
 
+extern jlib_decl bool isCloud()
+{
+    static bool checked = false;
```

Comment:
@richardkchapman 
Really really picky:  Both of these should be atomics.  Otherwise there is a small window on chips with weak memory ordering where another thread could read the value checked as true before it reads the updated value of cloudy.

Reply:
A static initialiser is guaranteed to be executed once, so maybe that would be cleaner(!!):

    static bool cloudy = []()
            {
                const char *env = getenv("HPCC_containerized");
                return (env && atoi(env)==1);
            }();
    return cloudy;

Reply:
repushed